src/styles/calendar.scss

```
@@ -0,0 +1,51 @@
+.calendar {
+  display: flex;
+  max-width: 730px;
+  padding: 10px;
+  flex-wrap: wrap;
+  gap: 1px;
+  margin: 0 auto;
+  font-family: Arial, Helvetica, sans-serif;
+
+  @each $day-name, $day in mon 0, tue 1, wed 2, thu 3, fri 4, sat 5, sun 6 {
+    &--start-day--#{$day-name} {
+      .calendar__day:nth-child(1) {
+        margin-left: 100px * $day + $day;
+      }
+    }
+  }
+
+  @for $day from 1 to 31 {
+    &--days-#{$day} {
+      .calendar__day:nth-child(n + #{$day + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    height: 100px;
+    width: 100px;
+    background-color: #eee;
+    border: 1px solid black;
+    display: flex;
+    font-size: 30px;
+    align-items: center;
+    justify-content: center;
+    transition: transform 500ms;
```

```suggestion
    transition: transform 500ms, background-color 500ms;
```
You can also specify transition to `background-color` and it changes smoothly too